calico/felix/ipsets.py

```diff
@@ -328,11 +314,18 @@ def remove_member(self, member):
 
     @actor_message()
     def on_unreferenced(self):
+        # Temporary set should never be left over; just in case, try to
+        # delete it.  Silence errors, we expect this to fail unless we're
+        # cleaning up after an error.
+        futils.call_silent(["ipset", "destroy", self.tmpname])
         try:
-            if self.set_exists:
+            if self.programmed_members is not None:
```

Comment:
Why not delete the set unconditionally? It seems to me that you could either get rid of the test and assume it exists (since it is benign to fail) or (better) test if the set exists by just calling ipset_exists. That feels more robust than figuring out whether the set exists based on whether you have written to it.